meson.build

```diff
@@ -0,0 +1,177 @@
+
+client_deserialice = custom_target('client-deserialice.c',
+                                   input: ['src/waltham/deserial-preamble.txt', protocols_src],
+                                   output: 'client-deserialice.c',
+                                   depend_files: 'tools/gen.py',
+                                   depends: client_h,
+                                   command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'client', '-t', 'demarshaller'])
+
+server_h = custom_target('waltham-server.h',
+                         input: ['src/waltham/header-preamble.txt', protocols_src],
+                         output: 'waltham-server.h',
+                         depend_files: 'tools/gen.py',
+                         command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'header'],
+                         install: true,
+                         install_dir: 'include/waltham')
+
+server_serialice = custom_target('server-serialice.c',
+                                 input: ['src/waltham/serial-preamble.txt', protocols_src],
+                                 output: 'server-serialice.c',
+                                 depend_files: 'tools/gen.py',
+                                 depends: server_h,
+                                 command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'marshaller'])
+
+server_deserialice = custom_target('server-deserialice.c',
+                                   input: ['src/waltham/deserial-preamble.txt', protocols_src],
+                                   output: 'server-deserialice.c',
+                                   depend_files: 'tools/gen.py',
+                                   depends: server_h,
+                                   command: [gen_py, '-p', '@INPUT0@', '-i', '@INPUT1@', '-i', '@INPUT2@', '-i', '@INPUT3@', '-o', '@OUTPUT@', '-m', 'server', '-t', 'demarshaller'])
+
+generated_src = [
+    client_h,
+    client_serialice,
+    client_deserialice,
+    server_h,
+    server_serialice,
+    server_deserialice,
+]
+
+add_global_arguments([
+    '-Wall',
+    '-Wextra',
+    '-Wno-format-zero-length',
+    '-Wno-unused-parameter',
+    '-Wstrict-prototypes',
+    '-Wmissing-prototypes',
+    '-Wno-missing-field-initializers',
+    '-fvisibility=hidden',
+], language: 'c')
+
+libwaltham_inc = include_directories('src/waltham')
+libwaltham = shared_library('waltham', libwaltham_src, generated_src,
+                            include_directories: libwaltham_inc,
+                            link_args: '-Wl,--no-undefined',
+                            install: true,
+                            version: version,
+                            soversion: version[0])
```

Comment:
I suspect one of `version` and `soversion` is redundant here.
I see that this installs the same names as autotools did, so let's keep it like that until we figure out the versioning again. I.e. hardcoding 0.0.0 if necessary is ok.